eda_analysis/eda_analysis.py

```diff
@@ -107,12 +111,66 @@ def describe_num_var(dataframe, num_vars):
     >>> X= pandas.DataFrame({
     'type':['Car', 'Bus', 'Car']
     'height':[10, 20, 30]
+    'width':[12, 15, 11]
     })
-    >>> num_vars = ['type']
+    >>> num_vars = ['height', 'width']
     >>> describe_num_var(X, num_vars)
       
     """
-    # Code
+    # Check the dataframe input
+    if not isinstance(dataframe, pd.DataFrame):
+        raise Exception("The value of the argument 'dataframe' should be of type pandas dataframe.")
+    
+    # Check the num_vars input should be a list of strings
+    if not (all(isinstance(item, str) for item in num_vars) & isinstance(num_vars, list)):
+        raise Exception("The value of the argument 'num_vars' should be a list of strings.")
+    
+    # Check if the elements in the num_vars input are unique
+    if len(num_vars) != len(set(num_vars)):
+        raise Exception("The elements in the argument 'num_vars' should be unique.")
+
+    # Check if the num_vars input contains only the column names
+    if not all(item in dataframe.columns for item in num_vars):
+        raise Exception("The argument 'num_vars' should be a subset of the column names of the dataframe.")
+    
+    # Subset and transpose the dataframe for later use 
+    df = pd.DataFrame(dataframe[num_vars]).T
+    
+    # Check if only the numeric columns are selected
+    if not np.issubdtype(df.to_numpy().dtype, np.number):
+        raise Exception("Only numeric columns expected, please check the input.")
+
+    # Calculate the statistical summaries
+    stat_funs = [np.nanmin, np.nanmax, np.nanmedian, np.nanmean, np.nanstd]
+    temp = []
+    for fun in stat_funs:
+        data_stat = df.apply(fun, axis=1)
+        temp.append(data_stat)
+
+    temp_df = pd.DataFrame(temp)
+    quantiles_df = pd.DataFrame([np.nanquantile(df, 0.25, axis = 1), 
+                                 np.nanquantile(df, 0.75, axis = 1)])
+    quantiles_df.columns = num_vars
+
+    summary = pd.concat([quantiles_df, temp_df])
+    
+    # Change the index more readable
+    summary.index = ["25%", "75%", "min", "max", "median", "mean", "sd"]
+    
+    # Make the histogram
+    plot = alt.Chart(df.T.melt().dropna()).mark_bar().encode(
+        alt.X("value:Q", bin=alt.Bin(maxbins=30), title = "Value"),
+        y='count()'
+    ).properties(
+        width=300,
+        height=300,
+        title='Histogram of Numeric Variables'
+    ).facet(
+        facet='variable:N',
+        columns=3
+        )
+    
+    return summary, plot
```

Comment:
When you return both of these would it print out the plot? When I tested it I only got the text with `(alt.FacetChart(...),`

Reply:
You are right, I need to update the example of how to use this function.
```
>>> summary, plot = describe_num_var(X, num_vars)
>>> summary
>>> plot
```